Assets/Scripts/Capabilities/Dash.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Dash : MonoBehaviour
+{
+    [SerializeField] InputController input = null;
+    [SerializeField] MovementOverride movement;
+    [SerializeField] Rigidbody2D rb;
+    [SerializeField] PlayerMovementController controller;
+    
+    private bool isDashing = false;
+    private Vector3 dashDirection;
+    private float dashSpeed;
+
+
+
+    void Update()
+    {
+        if (input.RetrieveDashInput())
+        {
+            isDashing = true;
+            dashDirection = Vector3.right;
+            if(!controller.IsFacingRight())
+            {
+                dashDirection = -dashDirection;
+            }
+            gameObject.layer = LayerMask.NameToLayer(Constants.INVINCIBLE_TAG);
```

Comment:
Nit - this is a layer, not a tag. I think we should change this to avoid confusion

```suggestion
            gameObject.layer = LayerMask.NameToLayer(Constants.INVINCIBLE_LAYER);
```